src/main/java/art/heredium/service/AccountService.java

```diff
@@ -328,4 +339,108 @@ public Page<AccountWithMembershipInfoIncludingTitleResponse> listWithMembershipI
       GetAccountWithMembershipInfoIncludingTitleRequest dto, Pageable pageable) {
     return accountRepository.searchWithMembershipInfoIncludingTitle(dto, pageable);
   }
+
+  @Transactional(rollbackFor = Exception.class)
+  public UploadCouponIssuanceTemplateResponse uploadCouponIssuance(MultipartFile file)
+      throws IOException {
+    UploadCouponIssuanceTemplateResponse response = new UploadCouponIssuanceTemplateResponse();
+    response.setCouponIssuanceAccounts(new ArrayList<>());
+
+    Set<String> processedIdentifiers = new HashSet<>();
+
+    Workbook workbook = WorkbookFactory.create(file.getInputStream());
+    Sheet sheet = workbook.getSheetAt(0);
+
+    // Skip header row and process each record
+    for (Row row : sheet) {
+      if (row.getRowNum() == 0) {
+        validateHeaderRow(row);
+        continue;
+      }
+      if (isRowEmpty(row)) continue;
+
+      String email = getCellValueAsString(row.getCell(0));
+      String phone = getCellValueAsString(row.getCell(1));
+
+      // Only check email for duplicates initially
+      boolean isDuplicateEmail =
+          StringUtils.isNotBlank(email) && processedIdentifiers.contains(email);
+      if (isDuplicateEmail) {
+        continue;
+      }
+
+      Account selectedAccount = null;
+
+      // Always try email first if present
+      if (StringUtils.isNotBlank(email)) {
+        Optional<Account> accountByEmail = accountRepository.findLatestLoginAccountByEmail(email);
+        if (accountByEmail.isPresent()) {
+          selectedAccount = accountByEmail.get();
+          processedIdentifiers.add(email);
+          String associatedPhone = selectedAccount.getAccountInfo().getPhone();
+          if (StringUtils.isNotBlank(associatedPhone)) {
+            processedIdentifiers.add(associatedPhone);
+          }
+          addAccountToResponse(selectedAccount, response);
+          continue;
+        }
+      }
+
+      // If no account found by email, try phone if not processed
+      if (StringUtils.isNotBlank(phone) && !processedIdentifiers.contains(phone)) {
```

Comment:
```suggestion
      if (!processedIdentifiers.contains(phone)) {
```
Phone blank ko thể có trong list.